src/Construction/DoctrineObjectConstructor.php

```diff
@@ -72,18 +86,50 @@ public function construct(DeserializationVisitorInterface $visitor, ClassMetadat
         $classMetadata = $objectManager->getClassMetadata($metadata->name);
         $identifierList = [];
 
+        /** @var \Doctrine\ORM\Mapping\Driver\AnnotationDriver $driver */
+        $driver = $objectManager->getConfiguration()->getMetadataDriverImpl();
+        if ($driver === null) {
+            throw new \RuntimeException(
+                'No cache driver implementation for the mapping metadata is given',
+                1526462942588
+            );
+        }
+        $reader = $driver->getReader();
+
         foreach ($classMetadata->getIdentifierFieldNames() as $name) {
+            $searchArray = $properties ?? $data;
+            $reflectionProperty = $classMetadata->getReflectionClass()->getProperty($name);
+            $isAttribute = (bool)$reader->getPropertyAnnotation(
```

Comment:
can not rely on this, people might be using yaml or xml metadatata, while this considers only annotations.

if you really want to look into the metadata, you have the `$metadata` for it

Reply:
Your right. I was blind since we only use Annotations as Metadata Source.